### ISSUE TITLE
panic: interface conversion: interface {} is string, not uint32

### ISSUE THREAD

**Issue by matperc** [1player/host-spawn#38]
```
panic: interface conversion: interface {} is string, not uint32

goroutine 1 [running]:
main.(*Command).waitForSignals(0xc0000c0120, 0xc00007c180)
        /home/runner/work/host-spawn/host-spawn/command.go:137 +0x1b5
main.(*Command).SpawnAndWait(0xc0000c0120)
        /home/runner/work/host-spawn/host-spawn/command.go:109 +0x997
main.main()
        /home/runner/work/host-spawn/host-spawn/main.go:121 +0x4fe
```

I use `host-spawn` to run commands from flatpak Emacs. Sometimes it crashes with this error. What could it be?

**Comment by matperc** [comment]
It's a complicated setup, I'll try to simplify a bit in the next days and see what happens.

Do I understand correctly that, even if the panic was handled, the command is still closing with HostCommandExited signal?

**Comment by matperc** [comment]
Ok I simplified my tries until this:
1. Download `host-spawn-x86-64` v1.6.0
2. Download emacs from Flathub
3. Run emacs
4. `M-x term` inside emacs (Alt-x term RET)
5. Run `host-spawn` without arguments and with absolute path (relative doesn't seems to work somehow)
6. Call some process for example `top`, or just wait
7. Some time later the error above appears, and the shell returns to flatpak

**Comment by 1player** [comment]
I cannot reproduce. From Flathub's Emacs (started with a bare config), within a term, host-spawn is able to spawn an host shell, top works correctly for me, etc. 

Can you provide any details about your linux distro? Can you reproduce the problem from inside a toolbox or distrobox container?

Your error seems to be caused by the fact that host-spawn, subscribed only to HostCommandExited DBus messages, receives an unexpected message and doesn't handle this case very well; it certainly should print a more helpful message than just panicking. I have seen weird errors from people using niche distros with missing or misconfigured flatpak/dbus.

If you are able to use git and compile go software, I can push some changes for you to test with.

**Comment by 1player** [comment]
Please try the `robustness` branch, which should print a more helpful error when host-spawn crashes.

https://github.com/1player/host-spawn/tree/robustness

**Comment by matperc** [comment]
Thanks for taking your time to look into it.

I failed to mention that the issue appears after a lot of time, randomly between 10 minutes to some hour. It seems worse when the PC is used intensively.

What do you mean inside a distrobox? Instead of flatpak?

I never tried compile a Go program, but I will try later today.

I replicated the issue both on Fedora Silverblue and Fedora Workstation. Both updated to the lastest version

**Comment by matperc** [comment]
```
unexpected DBus message &dbus.Signal{Sender:"org.freedesktop.DBus", Path:"/org/freedesktop/DBus", Name:"org.freedesktop.DBus.NameOwnerChanged", Body:[]interface {}{"org.freedesktop.portal.Flatpak", "", ":1.123"}, Sequence:0x5}
```

Here we go

**Comment by 1player** [comment]
Not sure what this message means, but I reckon host-spawn should check if it got the expected message, and ignored everything else. 

I'll work on a fix, thanks

On Tue, 12 Nov 2024, at 18:29, matperc wrote:
> 
> 
> `unexpected DBus message &dbus.Signal{Sender:"org.freedesktop.DBus", Path:"/org/freedesktop/DBus", Name:"org.freedesktop.DBus.NameOwnerChanged", Body:[]interface {}{"org.freedesktop.portal.Flatpak", "", ":1.123"}, Sequence:0x5}
`

**Comment by 1player** [comment]
Can you please try again with the latest commit on the `robustness` branch? Now we actually check the message name, and ignore anything that we don't care about.

**Comment by 1player** [comment]
Yes the fix is pretty straightforward, so I guess it won't crash again. I'll be merging into master and cutting a new release soon.

**Comment by 1player** [comment]
Fix released with version 1.6.1 (sorry for the delay)
